src/api/gigs.ts

```diff
@@ -1,143 +1,180 @@
-  const res = await runApi(
-    'POST',
-    `${API_URL}/contracts`,
-    body,
-    undefined,
-    true
-  );
-  return res;
+  const apiReqHeader = validateToken({}, true);
+
+  try {
+    const res = await axios.post(`${API_URL}/contracts`, body, apiReqHeader);
+    return res;
+  } catch (error: any) {
+    throw new Error(error.response.data.message);
+  }
 };
 
 export const getContractsForFreelancer = async (status?: string) => {
   const queryParams = status ? `?status=${status}` : '';
-  const res = await runApi(
-    'GET',
-    `${API_URL}/contracts/freelance${queryParams}`,
-    {},
-    {},
-    true
-  );
-  return res;
+  const apiReqHeader = validateToken({}, true);
+
+  try {
+    const res = await axios.get(
+      `${API_URL}/contracts/freelance${queryParams}`,
+      apiReqHeader
+    );
+    return res;
+  } catch (error: any) {
+    throw new Error(error.response.data.message);
+  }
 };
 
 export const getClientFreelancerOffers = async (
   freelancerUsername: string,
   clientUsername: string
 ) => {
   const reqBody = { freelancerUsername, clientUsername };
-  const res = await runApi(
-    'POST',
-    `${API_URL}/offers/between-participants`,
-    reqBody,
-    {},
-    true
-  );
-  return res;
+  const apiReqHeader = validateToken({}, true);
+
+  try {
+    const res = await axios.post(
+      `${API_URL}/offers/between-participants`,
+      reqBody,
+      apiReqHeader
+    );
+    return res;
+  } catch (error: any) {
+    throw new Error(error.response.data.message);
+  }
 };
 
 export const getContractForOffer = async (offerId: string) => {
-  console.log('offerId', offerId);
-  const res = await runApi(
-    'GET',
-    `${API_URL}/contracts/forOffer/${offerId}`,
-    {},
-    {},
-    true
-  );
-  return res;
+  const apiReqHeaders = validateToken({}, true);
+  try {
+    const res = await axios.get(
+      `${API_URL}/contracts/forOffer/${offerId}`,
+      apiReqHeaders
+    );
+    return res;
+  } catch (error: any) {
+    throw new Error(error.response.data.message);
+  }
 };
 
 export const updateContractStatus = async (
   contractId: string,
   status: string
 ) => {
   const reqBody = { status };
-  const res = await runApi(
-    'PATCH',
-    `${API_URL}/contracts/${contractId}/status`,
-    reqBody,
-    {},
-    true
-  );
-  return res;
+  const apiReqHeader = validateToken({}, true);
+
+  try {
+    const res = await axios.patch(
+      `${API_URL}/contracts/${contractId}/status`,
+      reqBody,
+      apiReqHeader
+    );
+    return res;
+  } catch (error: any) {
+    throw new Error(error.response.data.message);
+  }
 };
 
 export const updateMilestoneStatus = async (
   milestoneId: string,
   status: string
 ) => {
   const reqBody = { status };
-  const res = await runApi(
-    'PATCH',
-    `${API_URL}/milestones/${milestoneId}/status`,
-    reqBody,
-    {},
-    true
-  );
-  return res;
+  const apiReqHeader = validateToken({}, true);
```

Comment:
Suggest calling this **and all other occurrences** `reqHeader`. Is there any reason to include `api`?

```suggestion
  const reqHeader = validateToken({}, true);
```

Reply:
No specific reason other than following the prior naming conventions, I'll go ahead and change the occurrances to `reqHeader`!